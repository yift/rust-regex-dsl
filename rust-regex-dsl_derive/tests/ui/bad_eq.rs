pub use rust_regex_dsl::regex_dsl;

fn main() {
    let regex = regex_dsl!("");
    println!("{}", regex);
    let regex = regex_dsl!();
    println!("{}", regex);
    let regex = regex_dsl!(122);
    println!("{}", regex);
    let regex = regex_dsl!("test", "nop", 123);
    println!("{}", regex);
}