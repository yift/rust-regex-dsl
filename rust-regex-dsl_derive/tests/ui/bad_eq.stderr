error: Empty string is not supported
 --> tests/ui/bad_eq.rs:4:28
  |
4 |     let regex = regex_dsl!("");
  |                            ^^

error: Empty regex is not supported
 --> tests/ui/bad_eq.rs:6:17
  |
6 |     let regex = regex_dsl!();
  |                 ^^^^^^^^^^^^
  |
  = note: this error originates in the macro `regex_dsl` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected string literal or character literal
 --> tests/ui/bad_eq.rs:8:28
  |
8 |     let regex = regex_dsl!(122);
  |                            ^^^

error: expected string literal or character literal
  --> tests/ui/bad_eq.rs:10:43
   |
10 |     let regex = regex_dsl!("test", "nop", 123);
   |                                           ^^^
